pkg/api/apiloader.go

```diff
@@ -200,6 +200,10 @@ func (a *Apiloader) LoadContainerServiceForAgentPoolOnlyCluster(
 		if e := json.Unmarshal(contents, &managedCluster); e != nil {
 			return nil, IsSSHAutoGenerated, e
 		}
+		// verify managedCluster.Properties is not nil for creating case
+		if (managedCluster == nil || managedCluster.Properties == nil) && !hasExistingCS {
```

Comment:
`Properties` is a pointer. In update case if customer didn't set `Properties` would it be nil? Will that pointer need initialization in update case?

Reply:
Also add a test for this in: `apiloader_test.go`

Reply:
in update case, wouldn't nil `managedCluster.Properties` cause panic in line 208?